### DISCUSSION TITLE
website hosting changes

### DISCUSSION THREAD

**Discussion by hx2A** [py5coding/py5generator#148] [General]
@villares @marcovicci @tabreturn 

Very happy to share this with you today:

https://py5.readthedocs.io/en/latest/index.html

There are still some things that need to be straightened out before this should be shared on twitter, but they are all relatively minor compared to what I accomplished today. py5 requires a lot of customization for the readthedocs build to work correctly and it took a lot of experimentation to get right. I'll work through the other issues over the next week.

**Comment by hx2A** [comment]
OK, things are falling into place. I've fixed all of the critical readthedocs issues.

You'll notice there are two documentation versions on the website: "latest" and "dev". We can do PRs against dev to see how they look and then merge with the main branch to get them added to the "latest" version. We can have more versions (0.8.1, 0.8.2, etc), but I don't see the point in that kind of documentation granularity. We can always add that later if we feel we need it.

readthedocs has these webhooks whereby it will monitor the py5book repo and rebuild automatically when there is a PR. Neat!

There are two outstanding issues but neither is that important.

I'll issue a tweet about the new documentation website soon. I'm going to leave the existing website alone for now but will let people know it will be retired soonish. There will be a redirect or something sending visitors to the new location.

**Reply by villares** [reply]
I have lots of questions :)
- Will  **py5.ixora.io** redirect to readthedocs?  
- What should we use as the "project homepage" when mentioning the project?
- What should the "project homepage" contain? A minimal set of links like https://github.com/py5coding/?

**Reply by hx2A** [reply]
Yes, py5.ixora.io will redirect to py5.readthedocs.io. I think the new location should be considered the project home page. I think it is more professional than a subdomain of my personal website / blog. Although I'd be open to registering a py5 related domain name and linking it to readthedocs, which perhaps would be better still. Readthedocs supports adding your own domain name. I've looked at what is available for py5.<something> and nothing jumped out at me. [Can you have a look and let me know what you think?](https://domains.google.com/registrar/search?searchTerm=py5&hl=en&tab=1)

Readthedocs is a better and more reliable place to host documentation. I like how it can automatically rebuild documentation and manage versions.

Are you suggesting the py5 landing page https://py5.readthedocs.io/en/latest/ should have more links, or at least the links found on https://github.com/py5coding/? If so, that's a good point, we should add that to the main page.

**Reply by hx2A** [reply]
What is the page response time of py5.readthedocs.io as compared to py5.ixora.io? Since I am doing the hosting for the later, I know that it is set up in a way that will give fast page response times for anywhere in the world. I don't know if readthedocs is more US-centric or if it also works well for anywhere in the world. It is hard for me to evaluate this from a US location.

**Reply by hx2A** [reply]
Do you see any ads on the readthedocs pages? I don't see any but theoretically their free service is paid for with ads of some kind.

**Reply by hx2A** [reply]
Hmmm, readthedocs is serving pages via cloudflare. Existing website uses AWS cloudfront. Both are global CDNs and I imagine the performance difference between them is minimal.

BTW, I used the firefox extension [builtwith](https://addons.mozilla.org/en-US/firefox/addon/builtwith/) to see how readthedocs was serving their pages. That extension is very useful for examining websites.

**Reply by hx2A** [reply]
I registered a domain name and have it working correctly:

https://py5coding.org/

And I realized if we get a custom domain name and are not using multiple versions of the documentation in readthedocs website and if I am able to self-host, then readthedocs doesn't add much value in this case. With self-hosting I can access the web server logs and analyze them myself. There's no need for google analytics or any of the associated problems with that service.

**Reply by hx2A** [reply]
Things left to do:

- [x] add the py5-related links that are found on [github to the main landing page](https://github.com/py5coding/).
- [x] some reference docs like [`request_image()`](https://py5coding.org/reference/sketch_request_image.html) have links that can download resources from the new domain name.

Anything else I am missing?

**Reply by tabreturn** [reply]
Both domains/versions load fine and speedy for me --

![Screenshot from 2022-09-04 23-42-38](https://user-images.githubusercontent.com/9858145/188311598-9b5ba568-bc7c-4bd5-8259-c45f8412961e.png)

![Screenshot from 2022-09-04 23-42-44](https://user-images.githubusercontent.com/9858145/188311610-edd1a80b-20db-4133-a646-ba98d591bf36.png)

I don't see any ads on the readthedocs pages.

I just used the built-in dev tools (but builtwith looks interesting ...)

The new py5coding domain/site appears to work perfectly :)

**Reply by tabreturn** [reply]
One small thing -- the favicon doesn't load. EDIT: I don't think there is a favicon? The __static/logo.png_ could work for now?

**Reply by hx2A** [reply]
> One small thing -- the favicon doesn't load. EDIT: I don't think there is a favicon? The __static/logo.png_ could work for now?

Hmmm, I totally forgot about favicons. I believe they need to be a particular size? I'll shrink down the logo and see how it does.

**Reply by hx2A** [reply]
> Hmmm, I totally forgot about favicons. I believe they need to be a particular size? I'll shrink down the logo and see how it does.

That's taken care of. Almost done preparing the website.

**Comment by hx2A** [comment]
I've deleted the readthedocs documentation website in favor of the new website, http://py5coding.org.

I believe I have successfully updated all of the links that I can change. Some of the links, like the ones on [py5's pypi page](https://pypi.org/project/py5/), will change after the next release.

The existing http://py5.ixora.io website will stay is it is for now. After the next release I'll add a redirect to the new domain.
